include/eve/conditional.hpp

```diff
@@ -315,14 +361,28 @@ namespace eve
     static constexpr bool is_inverted     = false;
     static constexpr bool is_complete     = false;
 
-    constexpr keep_between_(std::ptrdiff_t b, std::ptrdiff_t e) noexcept : begin_(b), end_(e) {}
+    constexpr keep_between_(std::ptrdiff_t b, std::ptrdiff_t e) noexcept
+    : begin_(std::min(b,e)), end_(std::max(b,e))
+    {}
+
+    template<typename V> EVE_FORCEINLINE auto else_(V v) const  {  return or_(*this,v);  }
 
     template<typename T> EVE_FORCEINLINE auto mask(eve::as_<T> const&) const
     {
-      auto const i = detail::linear_ramp(eve::as_<T>());
+      auto const i = detail::linear_ramp(eve::as_<as_arithmetic_t<T>>());
       return (i >= begin_) && (i <= end_);
     }
 
+    template<typename T> EVE_FORCEINLINE std::ptrdiff_t offset(eve::as_<T> const&) const
+    {
+      return begin_;
+    }
+
+    template<typename T> EVE_FORCEINLINE auto count(eve::as_<T> const&) const
+    {
+      return sizeof(element_type_t<T>) * (end_ - begin_ + 1);
+    }
```

Comment:
+1? Why do you have a +1? Ignoring all elements is fine? What I can't keep_between 0 elements?

Reply:
Indeed, this is an error.